TESTS/mbed_platform/minimal-printf/compliance/main.cpp

```diff
@@ -845,44 +1066,71 @@ static control_t test_snprintf_f(const size_t call_count)
  * Template parameters:
     * 'T' is the type being tested
     * 'buf_size' is the buffer size used in tests
+    * 'radix_base' is used when Newlib-nano non-supported printf formats to calculate expected_string_length
   * Function parameters:
     * 'fmt' is the format to use for sprintf
     * 'data' is the data that will be printed
+    * 'sign' boolenan true for negative number,false for positive number
 */
+#if !defined(__NEWLIB_NANO)
 template<typename T, size_t buf_size>
-static control_t test_snprintf_buffer_overflow_generic(const char *fmt, T data)
+#else
+template<typename T, size_t buf_size, radix_base base = RADIX_10>
+#endif
+static control_t test_snprintf_buffer_overflow_generic(const char *fmt, T data, bool sign = false)
 {
     char buffer_baseline[buf_size];
     char buffer_minimal[buf_size];
     int result_baseline;
     int result_minimal;
+#if defined(__NEWLIB_NANO)
+    char prefix[buf_size];
+    memset(prefix, 0, buf_size);
+    extract_prefix(fmt, &prefix[0]);
```

Comment:
A comment here to explain why you extract prefix. This assumes that `fmt` is of a certain format so a word about that would be useful as well.

Reply:
You could have said: "Extract the string preceding the first '%'"

Reply:
// fmt string has "format specifier: %format specifier"
// Extract the string preceding the first '%' from fmt.